components/availability/src/Availability.svelte

```diff
@@ -889,28 +891,32 @@
 
     if (Array.isArray(calendarsToFetch) && calendarsToFetch.length > 0) {
       // Note: we split our requests by email_id as calendars/availability doesn't support partial availability amongst a group
-      return Promise.all(
-        calendarsToFetch.map(async (user) => {
-          const { email, token } = user;
-          let availableTimeslots = await $AvailabilityStore[
-            JSON.stringify({
-              ...getAvailabilityQuery([email], token ? token : access_token),
-              forceReload,
-            })
-          ];
 
-          // Blob consecutive timeslots together for easier overlap logic elsewhere
-          let groupedSlots = groupConsecutiveTimeslots(
-            availableTimeslots.time_slots,
+      let fetchedCalendars = await $AvailabilityStore[
+        JSON.stringify({
+          ...getAvailabilityQuery(
+            calendarsToFetch.map((x) => x.email),
+            access_token,
+          ),
+          forceReload,
+        })
+      ];
+      loading = false;
+
+      fetchedCalendars.order
+        .map((email) => {
+          const freeSlotsForUser = fetchedCalendars.time_slots.filter((ts) =>
```

Comment:
Small performance improvement here would be to avoid filtering inside the `map`. We could build out a map of users to time slots before we `.map()`, and then use that map here to lookup each user's timeslots. O(n+m) instead of O(n*m) complexity.